File: Projeto Didio Pizza/Didio Pizza Node/backend/src/controller/clienteController.js
import { Router } from 'express';
import * as clienteService from '../service/clienteService.js';
import { getAuthentication } from '../utils/jwt.js';

const auth = getAuthentication();

const endpoints = Router();


endpoints.post('/cliente', async (req, resp) => {
    try {
        let cliente = req.body;

        let id = await clienteService.criarCliente(cliente);

        resp.status(201).send({
            id: id
        });

    } catch (err) {
        resp.status(400).send({
            erro: err.message
        });
    }
});


endpoints.post('/cliente/login', async (req, resp) => {
    try {
        let cliente = req.body;

        let token = await clienteService.loginCliente(cliente);

        resp.send({
            token: token
        });

    } catch (err) {
        resp.status(401).send({
            erro: err.message
        });
    }
});


endpoints.get('/cliente/me', auth, async (req, resp) => {
    try {
        let clienteId = req.user.id;

        let dados = await clienteService.buscarPerfil(clienteId);

        resp.send(dados);

    } catch (err) {
        resp.status(400).send({ erro: err.message });
    }
});


endpoints.put('/cliente/me', auth, async (req, resp) => {
    try {
        let clienteId = req.user.id;

        let { nome, email, telefone } = req.body;

        await clienteService.atualizarPerfil(clienteId, nome, email, telefone);

        resp.status(204).send();

    } catch (err) {
        resp.status(400).send({ erro: err.message });
    }
});



export default endpoints;

File: Projeto Didio Pizza/Didio Pizza Springboot/backend/src/main/java/com/pizzaria/didiopizza/controllers/ClienteController.java
package com.pizzaria.didiopizza.controllers;

import java.util.Map;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestHeader;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestController;

import com.pizzaria.didiopizza.security.JwtUtil;
import com.pizzaria.didiopizza.service.ClienteService;

@RestController
public class ClienteController {
    private final ClienteService service;
    private final JwtUtil jwt;

    public ClienteController(ClienteService service, JwtUtil jwt) {
        this.service = service;
        this.jwt = jwt;
    }

    @PostMapping("/cliente")
    public ResponseEntity<?> criar(
            @RequestBody Map<String, Object> b) {
        return ResponseEntity.status(201).body(Map.of("id", service.criarCliente(b)));
    }

    @PostMapping("/cliente/login")
    public Map<String, Object> login(
            @RequestBody Map<String, Object> b) {
        return Map.of("token", service.loginCliente(b));
    }

    @GetMapping("/cliente/me")
    public Map<String, Object> me(
            @RequestHeader("Authorization") String auth) {
        return service.buscarPerfil(jwt.getId(auth));
    }

    @PutMapping("/cliente/me")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public void atualizar(
            @RequestHeader("Authorization") String auth,
            @RequestBody Map<String, Object> b) {
        service.atualizarPerfil(jwt.getId(auth), (String) b.get("nome"), (String) b.get("email"),
                (String) b.get("telefone"));
    }
}
